### PR TITLE
[13.0][FIX] delivery_auto_refresh: behave as core

### PR DESCRIPTION
**Pull request by chienandalu**
We should get as the same default carrier as Odoo does.

cc @Tecnativa TT34020

ping @CarlosRoca13 @pedrobaeza

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you point the code where Odoo does this?

Travis is red, and it seems this change is affecting tests of `delivery_multi_destination`. Maybe they become incompatible or there's an error on what to expect in the tests or lack of precision on the assigned data that leads to the problem. Please check.

**Comment by chienandalu** [comment]
Comment added and tests in `delivery_multi_destination` fixed to ensure test resilience

**Review by CarlosRoca13 — APPROVED** [review]

File: delivery_auto_refresh/models/sale_order.py [hunk 1]
```diff
@@ -13,14 +13,18 @@ class SaleOrder(models.Model):
...
-                if rec.partner_id.property_delivery_carrier_id:
-                    rec.carrier_id = rec.partner_id.property_delivery_carrier_id
+        # Compute the carrier as Odoo does in its core
+        # https://github.com/OCA/OCB/blob/27edc28a2c19fad80053e668a8c72e7a629e4c1a
+        # /addons/delivery/models/sale_order.py#L68-L71
+        for rec in self.filtered("partner_shipping_id"):
+            rec.carrier_id = (
+                rec.partner_shipping_id.property_delivery_carrier_id
+                or rec.partner_shipping_id.commercial_partner_id.property_delivery_carrier_id
+            ).filtered("active")
```
Comment (pedrobaeza) [inline delivery_auto_refresh/models/sale_order.py]: What do you think about calling the same method and extract the carrier and being compatible with possible extensions?

```suggestion
        for order in self:
            action = order.action_open_delivery_wizard()
            order.carrier_id = action["context"]["default_carrier_id"]
```

File: delivery_multi_destination/tests/test_delivery_multi_destination.py [hunk 2]
```diff
@@ -160,7 +160,8 @@ def test_delivery_multi_destination(self):
         sale_order_line = order.order_line.filtered("is_delivery")
         self.assertAlmostEqual(sale_order_line.price_unit, 50, 2)
         self.assertTrue(sale_order_line.is_delivery)
-        order.partner_shipping_id = self.partner_3.id
+        order.partner_shipping_id = self.partner_3
```
Comment (pedrobaeza) [inline delivery_multi_destination/tests/test_delivery_multi_destination.py]: Is modifying this line without the `.id` relevant for the test?

Comment (chienandalu) [reply delivery_multi_destination/tests/test_delivery_multi_destination.py]: No, it was a picky edit

Comment (pedrobaeza) [reply delivery_multi_destination/tests/test_delivery_multi_destination.py]: There are several of this FYI

File: delivery_auto_refresh/models/sale_order.py (hunk 1 above)
Comment (chienandalu) [reply delivery_auto_refresh/models/sale_order.py]: I like it. Let's go for it then

File: delivery_multi_destination/tests/test_delivery_multi_destination.py (hunk 2 above)
Comment (pedrobaeza) [reply delivery_multi_destination/tests/test_delivery_multi_destination.py]: I still see them. Please reduce the diff undoing these changes.

Comment (chienandalu) [reply delivery_multi_destination/tests/test_delivery_multi_destination.py]: Yeah, fixed

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**